pkg/watermark/fetch/edge_fetcher.go

```diff
@@ -33,10 +33,11 @@ func NewEdgeFetcher(ctx context.Context, edgeName string, processorManager *Proc
 	}
 }
 
-// GetHeadWatermark returns the watermark using the HeadOffset (latest offset). This
+// GetHeadWatermark returns the watermark using the HeadOffset (the latest offset among all processors). This
 // can be used in showing the watermark progression for a vertex when not consuming the messages
 // directly (eg. UX, tests,)
 func (e *edgeFetcher) GetHeadWatermark() processor.Watermark {
+	e.log.Debugf("Fetching head watermark for edge %s...", e.edgeName)
```

Comment:
Personally I would like to add as much meaningful logs as possible, making it easier to troubleshoot issues. Not sure if having a lot of logs is a bad thing in k8s's world or not...

Reply:
I agree; unfortunately, we do not have multiple levels of logging levels available (e.g., we do not have a verbose level). Since this is a high throughput system, we will be seeing a ton lot of logs just by moving the log-level to the next lowest level (i.e, Info to Debug). Also, the logs you have added are in the hot-code path, meaning they will be invoked for every message where the watermark operation is called. 

I am not proposing to lose important information but to use some tricks to make the logs more informational while reducing the number of log lines. E.g., there is a log entry to print the name of the edge, you could use [With](https://pkg.go.dev/go.uber.org/zap#Logger.With) to print out the edge for every log line.

Reply:
That's a good point. I didn't realize I was putting logs on a very hot code path. Will remove. Thanks!